Dp/5 build UI scaffold for student card on dashboard

DanielPurdy509 opened the pull request:
Student Card built it out with mock hard-coded data + some styling, added to dashboard
<img width="202" alt="Blue_Ocean_Student_Card_Component" src="https://github.com/gschool-blue-ocean/admissions-hub-20/assets/125397922/20309e92-2876-446e-93d5-4a833406bca6">

ranegray commented on client/src/components/StudentCard.jsx:
```diff
@@ -0,0 +1,71 @@
+                    <li className={StudentCardCSS.studentInfo}>Cohort: <b>{studentData.cohort}</b></li>
+                    <li className={StudentCardCSS.studentInfo}>Status: <b className={statusStyle}>{studentData.status}</b></li>
+                    <li className={StudentCardCSS.studentInfo}>Created By: <b>{studentData.createdBy}</b></li>
+                    <li className={StudentCardCSS.studentInfo}>Technical Interview Data: <b>{studentData.interviewDate}</b></li>
```
This should be technical interview date.

ranegray commented on client/src/components/StudentCard.jsx:
```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import StudentCardCSS from "../css/StudentCard.module.css";
```
Let's use object destructuring for cleaner JSX.

e.g. import { cardContainer, studentInfoCard, studentInfo, etc... } from <CSS FILE>

ranegray commented on client/src/components/StudentCard.jsx:
```diff
@@ -0,0 +1,71 @@
+
+    var statusStyle = "";
+
+    if(studentData.status == "Technical Interview"){
+        statusStyle = StudentCardCSS.interview;
+    }
+    if(studentData.status == "Coding Challenge"){
+        statusStyle = StudentCardCSS.challenge;
+    }
+    if(studentData.status == "Prep Work"){
+        statusStyle = StudentCardCSS.prep;
+    }
+    if(studentData.status == "Done"){
+        statusStyle = StudentCardCSS.done;
+    }
```
What if we made a status tag component that we can use on the card and in the table.

We can move this logic into that and all the CSS classes from the module to a new module for better reusability.

Let me know if you have any questions.

ranegray reviewed (changes requested):
As a general note, lets colocate CSS with our components. 

Create a new folder for StudentCard which will have StudentCard.jsx and StudentCard.module.css

There will be no CSS folder going forward.

jordan-carlson99 commented on client/src/components/StudentCard.jsx:
```diff
@@ -0,0 +1,71 @@
+        status: "Technical Interview",
+        createdBy: "Rane Gray",
+        interviewDate: "June 6th, 2023 @ 10:00AM MST",
+        notes: {
+            attemptNum: 1,
+            date: "May 23rd, 2023",
+            narrative: "Jon performed decently on most topics. Struggled with accessing properties on an object"
+        }
```
what would that look like for the database? csv? or should we consider making another table or column within users

DanielPurdy509 commented:
Summary of new changes:
- Reorganized components into folders with corresponding CSS files
- Created StatusTag component for use in the Student Card and elsewhere it is appropriate
- Refactored CSS stylings as requested 
- Changed the names of some of the student info properties as requested
